extension/src/plots/model/collect.ts

```diff
@@ -312,60 +342,131 @@ export const collectData = (output: PlotsOutput): DataAccumulator => {
   return acc
 }
 
-type ComparisonPlotsAcc = { path: string; revisions: ComparisonRevisionData }[]
+type ComparisonPlotsAcc = {
+  classDetails: ComparisonClassDetails
+  path: string
+  revisions: ComparisonRevisionData
+}[]
 
 type GetComparisonPlotImg = (
   img: ImagePlot,
   id: string,
   path: string
 ) => ComparisonPlotImg
 
+export const boundingBoxColors = [
+  '#ff3838',
+  '#ff9d97',
+  '#ff701f',
+  '#ffb21d',
+  '#cfd231',
+  '#48f90a',
+  '#92cc17',
+  '#3ddb86',
+  '#1a9334',
+  '#00d4bb',
+  '#2c99a8',
+  '#00c2ff',
+  '#344593',
+  '#6473ff',
+  '#0018ec',
+  '#8438ff',
+  '#520085',
+  '#cb38ff',
+  '#ff95c8',
+  '#ff37c7'
+]
+
+const collectComparisonPlotImgs = (
+  boundingBoxClassLabels: Set<string>,
+  imgs: ImagePlot[],
+  getComparisonPlotImg: GetComparisonPlotImg,
+  id: string,
+  path: string
+) => {
+  const plotImgs = []
+  for (const img of imgs) {
+    plotImgs.push(getComparisonPlotImg(img, id, path))
+
+    if (img.boundingBoxes) {
```

Comment:
This is a side effect of a function that returns something. It feels unexpected. `boundingBoxClassLabels` should be created inside this function and returned as well and recollected on the other side.

Reply:
Good catch! I've adjusted the code so that the class collection is in a separate function outside of the img collection to reduce complexity.